controllers/spifilecontentrequest_controller.go

```diff
@@ -154,88 +165,101 @@ func (r *SPIFileContentRequestReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, nil
 	}
 
-	if request.Status.Phase == "" {
-		// check if the URL is processable, otherwise fail fast
-		sp, _ := r.ServiceProviderFactory.FromRepoUrl(ctx, request.Spec.RepoUrl, request.Namespace)
-		if sp.GetDownloadFileCapability() == nil {
-			r.updateFileRequestStatusError(ctx, &request, serviceprovider.FileDownloadNotSupportedError{})
-			return ctrl.Result{}, serviceprovider.FileDownloadNotSupportedError{}
-		}
-		request.Status.Phase = api.SPIFileContentRequestPhaseAwaitingBinding
-	} else if request.Status.Phase == api.SPIFileContentRequestPhaseDelivered {
+	if request.Status.Phase == api.SPIFileContentRequestPhaseDelivered {
 		// already injected, nothing to do
 		return ctrl.Result{}, nil
 	}
+	// find associated binding
+	var associatedBinding *api.SPIAccessTokenBinding = nil
```

Comment:
I don't think you have to explicitly set it to `nil`, but nothing wrong with that